src/structures/Guild.js

```diff
@@ -1010,6 +1030,62 @@ class Guild extends Base {
       .then(newData => this.client.actions.GuildUpdate.handle(newData).updated);
   }
 
+  /**
+   * Welcome channel data
+   * @typedef {Object} WelcomeChannelData
+   * @property {string} description The description to show for this welcome channel
+   * @property {GuildChannelResolvable} channel The channel to link for this welcome channel
+   * @property {EmojiIdentifierResolvable} [emoji] The emoji to display for this welcome channel
+   */
+
+  /**
+   * Welcome screen edit data
+   * @typedef {Object} WelcomeScreenEditData
+   * @property {boolean} [enabled] Whether the welcome screen is enabled
+   * @property {string} [description] The description for the welcome screen
+   * @property {WelcomeChannelData[]} [welcomeChannels] The welcome channel data for the welcome screen
+   */
+
+  /**
+   * Updates the guild's welcome screen
+   * @param {WelcomeScreenEditData} [data] Data to edit the welcome screen with
+   * @returns {Promise<WelcomeScreen>}
+   * @example
+   * guild.editWelcomeScreen({
+   *   description: 'Hello World',
+   *   enabled: true,
+   *   welcomeChannels: [
+   *     {
+   *       description: 'foobar',
+   *       channel: '222197033908436994',
+   *     }
+   *   ],
+   * })
+   */
+  editWelcomeScreen({ enabled, description, welcomeChannels } = {}) {
+    const welcome_channels =
+      welcomeChannels?.map(data => {
+        const channel_id = this.channels.resolveID(data.channel);
+        const emoji = this.emojis.resolve(data.emoji);
+        return {
+          emoji_id: emoji?.id ?? null,
+          emoji_name: emoji?.name ?? emoji,
+          channel_id,
+        };
+      }) ?? undefined;
```

Comment:
The optional chaining would make it return `undefined` anyways
```suggestion
    const welcome_channels = welcomeChannels?.map(data => {
      const emoji = this.emojis.resolve(data.emoji);
      return {
        emoji_id: emoji?.id ?? null,
        emoji_name: emoji?.name ?? emoji,
        channel_id: this.channels.resolveID(data.channel),
      };
    });
```